src/frontends/ir/tests/frontend_test.cpp

```diff
@@ -0,0 +1,1213 @@
+            </input>
+        </layer>
+    </layers>
+    <edges>
+        <edge from-layer="0" from-port="0" to-layer="2" to-port="0"/>
+        <edge from-layer="1" from-port="0" to-layer="2" to-port="1"/>
+        <edge from-layer="2" from-port="2" to-layer="3" to-port="0"/>
+    </edges>
+</net>
+)V0G0N";
+
+    ASSERT_THROW((void)core.read_model(testNetworkV11, ov::Tensor()), ov::Exception);
+}
+
+TEST_F(IrFrontendTests, NetworkWithWeightsFromDisk) {
+    std::string xmlModel = R"V0G0N(
+<?xml version="1.0" ?>
+<net name="Network" version="11">
+    <layers>
+        <layer name="input" type="Parameter" id="0" version="opset1">
+            <data element_type="f32" shape="1,3,22,22"/>
+            <output>
+                <port id="0" precision="FP32">
+                    <dim>1</dim>
+                    <dim>3</dim>
+                    <dim>22</dim>
+                    <dim>22</dim>
+                </port>
+            </output>
+        </layer>
+        <layer id="1" name="value1" type="Const" version="opset1">
+            <data element_type="i64" shape="4" offset="0" size="32" />
+            <output>
+                <port id="0" precision="I64">
+                    <dim>4</dim>
+                </port>
+            </output>
+        </layer>
+        <layer id="2" name="Transpose0321" type="Transpose" version="opset1">
+            <input>
+                <port id="0" precision="FP32">
+                    <dim>1</dim>
+                    <dim>3</dim>
+                    <dim>22</dim>
+                    <dim>22</dim>
+                </port>
+                <port id="1" precision="I64">
+                    <dim>4</dim>
+                </port>
+            </input>
+            <output>
+                <port id="2" precision="FP32">
+                    <dim>1</dim>
+                    <dim>22</dim>
+                    <dim>22</dim>
+                    <dim>3</dim>
+                </port>
+            </output>
+        </layer>
+        <layer name="output" type="Result" id="3" version="opset1">
+            <input>
+                <port id="0" precision="FP32">
+                    <dim>1</dim>
+                    <dim>22</dim>
+                    <dim>22</dim>
+                    <dim>3</dim>
+                </port>
+            </input>
+        </layer>
+    </layers>
+    <edges>
+        <edge from-layer="0" from-port="0" to-layer="2" to-port="0"/>
+        <edge from-layer="1" from-port="0" to-layer="2" to-port="1"/>
+        <edge from-layer="2" from-port="2" to-layer="3" to-port="0"/>
+    </edges>
+</net>
+)V0G0N";
+
+    std::vector<char> buffer(32, 0);
+    uint64_t* uint64Buffer = reinterpret_cast<uint64_t*>(buffer.data());
+    uint64Buffer[0] = 0;
+    uint64Buffer[1] = 3;
+    uint64Buffer[2] = 2;
+    uint64Buffer[3] = 1;
+
+    createTemporalModelFile(xmlModel, buffer);
+
+    std::shared_ptr<ov::Model> model;
+
+    ASSERT_NO_THROW(model = core.read_model(xmlFileName, binFileName));
```

Comment:
I think will be better also to check that model contains the right number of operations and right operations